packages/contracts/scripts/test_userOp_testnet.ts

```diff
@@ -147,41 +149,98 @@ async function testConsecutiveUserOps(kernelAccount: SmartAccount, kernelClient:
 
     console.log("     Deployed successfully")
     console.log("\n[2/3] Sending UserOps...")
-    const hashes = []
+    const userOpsAndHashes = []
     const currentNonce = await kernelClient.account?.getNonce()
     if(!currentNonce){
         throw new Error("Failed to get nonce")
     }
     console.log("     Current nonce:", currentNonce)  
-    for(let i = 0; i < NUMBER_OF_USEROPS; i++){
-        const hash = await sendUserOp(kernelAccount, kernelClient, currentNonce + BigInt(i))
-        hashes.push(hash)
+    for(let i = 0; i < numberOfUserOps; i++){
+        const res = await sendUserOpNow(kernelAccount, kernelClient, currentNonce + BigInt(i))
+        userOpsAndHashes.push(res)
         
     }
-    console.log("\n[3/3] Awaiting receipts...")
+    // console.log("\n[3/3] Awaiting receipts...")
 
-    const statuses = await Promise.all(hashes.map(hash => pollForCompletion(hash)))
+    // const statuses = await Promise.all(userOpsAndHashes.map(userOpAndHash => pollForCompletion(userOpAndHash.hash)))
 
-    for(let i = 0; i < statuses.length; i++){
-        console.log(`UserOp ${i} status: ${statuses[i]}`)
-    }
-    console.log("Completed sending and awaiting receipts")
+    // for(let i = 0; i < statuses.length; i++){
+    //     fs.appendFileSync("userOpTimingMap.txt", `UserOp ${userOpsAndHashes[i].hash} status: ${statuses[i]}, time taken: ${userOpTimingMap.get(userOpsAndHashes[i].hash)?.finalStatusTime! - userOpTimingMap.get(userOpsAndHashes[i].hash)?.sendTimestamp!}ms\n`)
+    //     // console.log(`UserOp ${userOpsAndHashes[i].hash} status: ${statuses[i]}, time taken: ${userOpTimingMap.get(userOpsAndHashes[i].hash)?.finalStatusTime! - userOpTimingMap.get(userOpsAndHashes[i].hash)?.sendTimestamp!}ms`)
+    // }
+    // console.log("Completed sending and awaiting receipts")
 }
 
-async function sendUserOp(kernelAccount: SmartAccount, kernelClient: SmartAccountClient, nonce: bigint){
+async function sendUserOpNow(kernelAccount: SmartAccount, kernelClient: SmartAccountClient, nonce: bigint){
+    const timeStart = Date.now()
     const userOp: UserOperation<"0.7"> = await kernelClient.prepareUserOperation({
         account: kernelAccount,
         calls: [createTransferCall(getRandomAccount())],
     })
+    const strippedUserOp = {
```

Comment:
"just" the onchain UserOp struct needs to be passed, the returned "userOp" from `kernelClient.prepareUserOperation()` has a bunch of other stuff in it